cupy/sorting/search.py

```diff
@@ -181,7 +182,132 @@ def where(condition, x=None, y=None):
     return _where_ufunc(condition.astype('?'), x, y)
 
 
-# TODO(okuta): Implement searchsorted
+# This is to allow using the same kernels for all dtypes, ints & floats
+# as nan is a special case
+_preamble = '''
+template<typename T>
+__device__ bool _isnan(T val) {
+    return false;
+}
+template<>
```

Comment:
These `template<>` should be removed.

Reply:
it didnt work without it, as the kernel code uses the template everything resorted to false, if i remove the template instantiation from the kernel then it cant resolve the integral types 😥

Reply:
OK, thanks.

Reply:
please let me know if you find a cleaner way than this, I dont like it as well ...

Reply:
please let me know if you find a cleaner way than this, I dont like it as well ...